CMakeLists.txt

```diff
@@ -1034,6 +1034,10 @@ if(${PLAYER_BUILD_EXECUTABLE} AND ${PLAYER_TARGET_PLATFORM} STREQUAL "SDL2")
 	target_link_libraries(${EXE_NAME} ${PROJECT_NAME})
 
 	if(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
+		# Do not confuse this with our own -DUSE_SDL
+		# Prevent Emscripten from pulling in bundled SDL headers that break the build
+		target_compile_options(${PROJECT_NAME} PUBLIC "-sUSE_SDL=0")
+
```

Comment:
Minor style nit: we us all other `-s` options with added space (see line 1054 for example)

`src/platform/sdl/main.cpp` pulls in SDL headers as well, we might want to force this there (`${EXE_NAME}`) in the same way. Or just use `add_definitions()`.

Reply:
Oh, wait. `PUBLIC` does this already.

Reply:
The ``-sUSE_SDL=0`` is also provided to ``EXE_NAME`` because it is a ``PUBLIC`` compile option. But I will fix the style nit :)

```
buildscripts/emscripten/emsdk-portable/upstream/emscripten/em++
    -DHAVE_FLUIDSYNTH=1
    -DHAVE_FREETYPE=1
    -DHAVE_LIBMPG123=1
    -DHAVE_LIBSNDFILE=1
    -DHAVE_LIBSPEEXDSP=1
    -DHAVE_LIBXMP=1
    -DHAVE_OGGVORBIS=1
    -DHAVE_OPUS=1
    -DHAVE_WINE=1
    -DSUPPORT_AUDIO=1
    -DUSE_SDL=2
    -DWANT_DRWAV=1
    -DWANT_FMMIDI=1
    -Ibuildscripts/emscripten/easyrpg-player/src
    -isystem
    buildscripts/emscripten/include/SDL2
    -isystem
    buildscripts/emscripten/include
    -isystem
    buildscripts/emscripten/lib
    -isystem
    buildscripts/emscripten/include/pixman-1
    -isystem
    buildscripts/emscripten/include/freetype2
    -isystem
    buildscripts/emscripten/include/speex
    -isystem
    buildscripts/emscripten/include/vorbis
    -isystem
    buildscripts/emscripten/include/opus
    -isystem
    buildscripts/emscripten/include/libxmp-lite
    -fno-math-errno
    -sUSE_SDL=0   <----
    -std=gnu++14
    -MD
    -MT
    CMakeFiles/EasyRPG_Player_exe.dir/src/platform/sdl/main.cpp.o
    -MF
    CMakeFiles/EasyRPG_Player_exe.dir/src/platform/sdl/main.cpp.o.d
    -o
    CMakeFiles/EasyRPG_Player_exe.dir/src/platform/sdl/main.cpp.o
    -c
    buildscripts/emscripten/easyrpg-player/src/platform/sdl/main.cpp
```

Reply:
This is weird: When I use ``"-s USE_SDL=0"`` CMake will append it with the quotes included so the build will fail. What works is ``"-s;USE_SDL=0"``.

No idea why ``LINK_FLAGS`` does not have this special treatment though o_O

Reply:
In this case, let's keep the current way (no space).